<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to <code>badirater</code> • badirater</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to &lt;code&gt;badirater&lt;/code&gt;">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">badirater</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/badirater.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to <code>badirater</code>
</h1>
                        <h4 class="author">Gergo Palfalvi</h4>
            <address class="author_afil">
      National Institute for Basic Biology, Japan<br><a class="author_email" href="mailto:#"></a><a href="mailto:palfalvi.gergo@gmail.com">palfalvi.gergo@gmail.com</a>
      </address>
                  
            <h4 class="date">2018-08-16</h4>
      
      
      <div class="hidden name"><code>badirater.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="what-is-badirater" class="section level1">
<h1 class="hasAnchor">
<a href="#what-is-badirater" class="anchor"></a>What is <code>badirater</code>?</h1>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>badirater</code> is an R package developed to make genomic studies easier using family turnover rate estimations with <a href="http://www.ub.edu/softevol/badirate/"><code>BadiRate</code></a>. It enables genome wide studies along phylogenetic trees from within R, including data preparation, help for execution and downstream processing.</p>
</div>
<div id="how-badirate-works-in-whole-genome-level-studies" class="section level2">
<h2 class="hasAnchor">
<a href="#how-badirate-works-in-whole-genome-level-studies" class="anchor"></a>How BadiRate works in whole genome level studies?</h2>
<p><a href="http://www.ub.edu/softevol/badirate/"><code>BadiRate</code></a> is a versatile program which enables family turnover rate analysis, including gene family expansion and collapsion detections, however it’s structure and style makes it difficult to work with in large-scale studies.</p>
<p>Standard method includes at least one run per family/orthogroup (og). This will be expanded by comparison of different branch models and technical replicates. Formerly is necessary to infer model estimation accuracy, later is to avoid local maxima detection instead of global one. If we consider only 3 replications with 3 different models on ~10.000 gene families, it is clear that we should run BadiRate ~90.000 times. This can be a reason for the inexperienced users to leave the program and look for similar, but less versatile ones.</p>
<p><code>badirater</code> enables to automate the whole process, including data preprocessing, batch run preparation, data collection and tidying and downstream calculations and processes.</p>
</div>
</div>
<div id="requirements" class="section level1">
<h1 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h1>
<div id="operating-system" class="section level2">
<h2 class="hasAnchor">
<a href="#operating-system" class="anchor"></a>Operating system</h2>
<p><code>badirater</code> uses reguler UNIX features, e.g. <code>awk</code>, hence it is required to be used on UNIX based systems. Since BadiRate is a perl program and we expect batch processing, it is recommended to process on an available cluster computer with Torque/PBS or SLURM.</p>
</div>
<div id="required-programs" class="section level2">
<h2 class="hasAnchor">
<a href="#required-programs" class="anchor"></a>Required programs</h2>
<p>To run <code>badirater</code>, the following programs should be installed:</p>
<ul>
<li><a href="http://www.ub.edu/softevol/badirate/"><code>BadiRate</code></a></li>
</ul>
</div>
<div id="input-files" class="section level2">
<h2 class="hasAnchor">
<a href="#input-files" class="anchor"></a>Input files</h2>
<p>There are just two input files are required:</p>
<ul>
<li>Gene family sizes: either direct output from <code>orthofinder</code> (Orthogroups.GeneCount.csv) or a similarly formatted file.</li>
<li>Ultrametric species tree</li>
</ul>
<p>Also you have to consider the branch models you would like to test. Regularly we run at least three different models: a global rate, a free rate and a species-specific rate model:</p>
<ul>
<li>Global rate (GR): All branches have the same turnover rate</li>
<li>Free rate (FR): All branches have independent turnover rates</li>
<li>Branch specific (BR): The user supplies which branches are grouped as common turnover rate (see BadiRate documentation for further explanation)</li>
</ul>
</div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages('devtools')</span>
devtools::<span class="kw">install_github</span>(<span class="st">"palfalvi/badirater"</span>)</code></pre></div>
</div>
<div id="preparing-for-badirate-run" class="section level1">
<h1 class="hasAnchor">
<a href="#preparing-for-badirate-run" class="anchor"></a>Preparing for BadiRate run</h1>
<p>First let’s arrange our working directory as follows:</p>
<pre><code>- badirate_analysis/
  |- Orthogroups.GeneCount.csv
  |- SpeciesTree.Ultrametric.tree</code></pre>
<div id="splitting-family-size-file" class="section level2">
<h2 class="hasAnchor">
<a href="#splitting-family-size-file" class="anchor"></a>Splitting family size file</h2>
<p>Since we would like to run BadiRate by gene family/orthogroup, we should split the files into individual components. All the new files will be stored in one directory.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(badirater)

<span class="kw">setwd</span>(<span class="st">"../test/"</span>)

<span class="kw"><a href="../reference/split_orthogroups.html">split_orthogroups</a></span>(<span class="st">"../test/obp_sub.12sp.tsv"</span>, <span class="dt">max_count =</span> <span class="dv">40</span>, <span class="dt">og_path =</span> <span class="st">"../test/badirate_orthogroups"</span>)</code></pre></div>
<p>This created the following:</p>
<pre><code>- badirate_analysis/
  |- badirate_orthogroups/
      |- OGxxxxxx.txt
      |- ...
      |- OGyyyyyy.txt
  |- Orthogroups.GeneCount.csv
  |- SpeciesTree.Ultrametric.tree</code></pre>
</div>
<div id="creating-job-scripts-and-setup-table" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-job-scripts-and-setup-table" class="anchor"></a>Creating job scripts and setup table</h2>
<p>Now we have all the family files prepared, we can set up the experiment for BadiRate. Create the branch models object you would like to select, and set up the script files.</p>
<p>perl BadiRate.pl –treefile examples/droso.6sp.tamura.nwk –sizefile examples/obp_all.12sp.tsv –anc</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bd_path =<span class="st"> "/Users/gergo/Downloads/badirate-1.35/"</span>

<span class="kw">branch_numbers</span>(<span class="dt">tree =</span> <span class="st">"../test/droso.12sp.tamura.nwk"</span>,
               <span class="dt">og =</span> <span class="st">"../test/obp_sub.12sp.tsv"</span>,
               <span class="dt">badirate_path =</span> bd_path,
               <span class="dt">plot =</span> <span class="ot">TRUE</span>,
               <span class="dt">tree_file =</span> <span class="st">"../test/branch_ids.tree"</span>)</code></pre></div>
<p>Now we see the branch numbering and can describe branch models. Important, that each branch model should have an exactly 2-letter long unique name. In this case, we prepare 4 models, one global rate and one free rate model and 2 branch specific ones (<code>sp</code> and <code>fm</code>), both with one group and background.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">branch_models =<span class="st"> </span><span class="kw">c</span>(
  <span class="st">"gr"</span> =<span class="st"> "GR"</span>,
  <span class="st">"sp"</span> =<span class="st"> "22-&gt;21"</span>
)</code></pre></div>
<p>As we decided our models and every files are ready, we can prepare the running scripts. Don’t forget to save the output of the <code><a href="../reference/prepare_badirate.html">prepare_badirate()</a></code> for future use. If you forgot, still you can rerun with <code>create_scripts = "none"</code> option.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">setup_table &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepare_badirate.html">prepare_badirate</a></span>(<span class="dt">og_path =</span> <span class="st">"./badirate_orthogroups/"</span>,
                                <span class="dt">tree =</span> <span class="st">"./droso.12sp.tamura.nwk"</span>,
                                <span class="dt">branch_models =</span> branch_models,
                                <span class="dt">rate_model =</span> <span class="st">"GD"</span>,
                                <span class="dt">out_dir =</span> <span class="st">"./raw_outputs"</span>,
                                <span class="dt">script_dir =</span> <span class="st">"./scripts"</span>,
                                <span class="dt">replicates =</span> <span class="dv">2</span>,
                                <span class="dt">ancestral =</span> <span class="ot">TRUE</span>,
                                <span class="dt">outlier =</span> <span class="ot">TRUE</span>,
                                <span class="dt">seed =</span> <span class="dv">20180808</span>,
                                <span class="dt">start_value =</span> <span class="dv">1</span>,
                                <span class="dt">pbs_q =</span> <span class="st">"small"</span>,
                                <span class="dt">badirate_path =</span> bd_path,
                                <span class="dt">create_scripts =</span> <span class="st">"pbs"</span>)


setup_table</code></pre></div>
<p>Don’t forget to save the <code>setup_table</code> object!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">readr::<span class="kw">write_csv</span>(setup_table, <span class="st">"./setup_table.csv"</span>)</code></pre></div>
<p>Now our working directory should look like this:</p>
<pre><code>- badirate_analysis/
  |- raw_outputs/
  |- scripts/
      |- badi_gr_script.pbs
      |- badi_fr_script.pbs
      |- badi_sp_script.pbs
      |- badi_fm_script.pbs
  |- badirate_orthogroups/
      |- OGxxxxxx.txt
      |- ...
      |- OGyyyyyy.txt
  |- setup_table.csv
  |- Orthogroups.GeneCount.csv
  |- SpeciesTree.Ultrametric.tree</code></pre>
</div>
</div>
<div id="running-badirate-jobs" class="section level1">
<h1 class="hasAnchor">
<a href="#running-badirate-jobs" class="anchor"></a>Running BadiRate jobs</h1>
<p>There are several ways you can run the created jobs in the <code>./scripts</code> directory. If your system has available PBS or SLURM, you could create ready-to-run scripts with <code>preapre_badirate()</code>, otherwise you can use those scripts as a starting point and run in your command line directly <a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a>.</p>
<p>In <code>badirater</code> there are some built in function to be able to submit and monitor PBS and SLURM jobs without leaving R.</p>
<div id="pbstorque" class="section level2">
<h2 class="hasAnchor">
<a href="#pbstorque" class="anchor"></a>PBS/Torque</h2>
<p>To submit jobs, use <code><a href="../reference/qsub.html">qsub()</a></code> with a file or filelist. You can monitor the run with <code><a href="../reference/qstat.html">qstat()</a></code> or <code>qstat("-Jt"")</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scripts &lt;-<span class="st"> </span><span class="kw">dir</span>(<span class="dt">path =</span> <span class="st">"./scripts"</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)

<span class="kw"><a href="../reference/qsub.html">qsub</a></span>(scripts)

<span class="kw"><a href="../reference/qstat.html">qstat</a></span>()</code></pre></div>
</div>
<div id="slurm" class="section level2">
<h2 class="hasAnchor">
<a href="#slurm" class="anchor"></a>SLURM</h2>
<p>To submit jobs, use <code><a href="../reference/sbatch.html">sbatch()</a></code> with a file or filelist. You can monitor the run with <code><a href="../reference/squeue.html">squeue()</a></code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scripts &lt;-<span class="st"> </span><span class="kw">dir</span>(<span class="dt">path =</span> <span class="st">"./scripts"</span>)

<span class="kw"><a href="../reference/sbatch.html">sbatch</a></span>(scripts)

<span class="kw"><a href="../reference/squeue.html">squeue</a></span>()</code></pre></div>
</div>
</div>
<div id="collecting-outputs" class="section level1">
<h1 class="hasAnchor">
<a href="#collecting-outputs" class="anchor"></a>Collecting outputs</h1>
<p>After all the jobs finished (may take a while), it is time to collect the important data from each files in the <code>./raw_outputs</code> directory.</p>
<pre><code>- badirate_analysis/
  |- raw_outputs/
      |- gr/
          |- OGxxxxxx.txt.gr01.bd
          |- ...
      |- fr/
          |- OGxxxxxx.txt.fr01.bd
          |- ...
      |- ...
  |- scripts/
      |- badi_gr_script.pbs
      |- badi_fr_script.pbs
      |- badi_sp_script.pbs
      |- badi_fm_script.pbs
  |- badirate_orthogroups/
      |- OGxxxxxx.txt
      |- ...
      |- OGyyyyyy.txt
  |- setup_table.csv
  |- Orthogroups.GeneCount.csv
  |- SpeciesTree.Ultrametric.tree</code></pre>
<p>Use the <code><a href="../reference/bd_collect.html">bd_collect()</a></code> function for automatic collection. It may take a while, depending on the volume of the experiment.</p>
<p>This will collect 4 files in each branch model - replication group:</p>
<ul>
<li>
<code>nnXX.branch_code.txt</code>: Branch decodings</li>
<li>
<code>nnXX.likelihood.txt</code>: Likelihood values for each families/orthogroups</li>
<li>
<code>nnXX.parameters.txt</code>: Calculated turnover rates</li>
<li>
<code>nnXX.gains.txt</code>: Minimum gain and loss information per branch</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">setup_table &lt;-<span class="st"> </span>readr::<span class="kw">read_csv</span>(<span class="st">"../test/setup_table.csv"</span>)

<span class="kw"><a href="../reference/bd_collect.html">bd_collect</a></span>(setup_table, <span class="dt">out_dir =</span> <span class="st">"../test/results"</span>)</code></pre></div>
<pre><code>- badirate_analysis/
  |- results/
      |- grXX.branch_code.txt
      |- grXX.likelihood.txt
      |- grXX.parameters.txt
      |- grXX.gains.txt
      |- ...
  |- raw_outputs/
      |- gr/
          |- OGxxxxxx.txt.gr01.bd
          |- ...
      |- fr/
          |- OGxxxxxx.txt.fr01.bd
          |- ...
      |- ...
  |- scripts/
      |- badi_gr_script.pbs
      |- badi_fr_script.pbs
      |- badi_sp_script.pbs
      |- badi_fm_script.pbs
  |- badirate_orthogroups/
      |- OGxxxxxx.txt
      |- ...
      |- OGyyyyyy.txt
  |- setup_table.csv
  |- Orthogroups.GeneCount.csv
  |- SpeciesTree.Ultrametric.tree</code></pre>
</div>
<div id="model-selection-and-gainloss-estimates" class="section level1">
<h1 class="hasAnchor">
<a href="#model-selection-and-gainloss-estimates" class="anchor"></a>Model selection and gain/loss estimates</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="st">"./analysis"</span>)</code></pre></div>
<div id="selecting-best-models-and-testing-its-significance" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-best-models-and-testing-its-significance" class="anchor"></a>Selecting best models and testing its significance</h2>
<p>Here we use AIC (Akaike Information Criteria) to select the best replicates per model per family and wAIC (weighted AIC) to select the best models per orthogroup and assest the model significance.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">best_models &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bd_model_select.html">bd_model_select</a></span>(<span class="dt">setup_table =</span> setup_table, 
                               <span class="dt">results_dir =</span> <span class="st">"./results"</span>)


readr::<span class="kw">write_csv</span>(best_models, <span class="st">"./analysis/best_models.csv"</span>)

best_models</code></pre></div>
<p>The output is tibble with one line per family, with the selected best model, its AIC and wAIC number, and its wAIC ratio to the second best model (it is accepted as significant if wAIC ratio is greater then 2.7, which means log(2.7) = 0.99325 is the probability that we selected a more valid model). This is corresponding to the <code>signif</code> column.</p>
</div>
<div id="parsing-gainloss-events" class="section level2">
<h2 class="hasAnchor">
<a href="#parsing-gainloss-events" class="anchor"></a>Parsing gain/loss events</h2>
<p>We can parse gain/loss events for each orthogroup from the best models.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gains &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bd_extract_gain.html">bd_extract_gain</a></span>(<span class="dt">best_models =</span> best_models, <span class="dt">results_dir =</span> <span class="st">"./results"</span>)

gains</code></pre></div>
<p>Now we have most of the information on significant gain/loss events.</p>
<hr>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<p>This document was generated on the following system:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.1 (2017-06-30)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] C
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] BiocStyle_2.4.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.18     rstudioapi_0.7   knitr_1.20       xml2_1.1.1      
##  [5] magrittr_1.5     roxygen2_6.1.0   MASS_7.3-47      R6_2.2.2        
##  [9] rlang_0.2.1      stringr_1.3.1    tools_3.4.1      htmltools_0.3.6 
## [13] commonmark_1.5   yaml_2.2.0       rprojroot_1.2    digest_0.6.15   
## [17] assertthat_0.2.0 pkgdown_1.1.0    crayon_1.3.4     fs_1.2.5        
## [21] memoise_1.1.0    evaluate_0.11    rmarkdown_1.6    stringi_1.2.4   
## [25] compiler_3.4.1   desc_1.2.0       backports_1.1.2</code></pre>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Too complex or too many jobs can run for days. Be careful to save everything if you leave the computer.<a href="#fnref1">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#what-is-badirater">What is <code>badirater</code>?</a><ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li><a href="#how-badirate-works-in-whole-genome-level-studies">How BadiRate works in whole genome level studies?</a></li>
      </ul>
</li>
      <li>
<a href="#requirements">Requirements</a><ul class="nav nav-pills nav-stacked">
<li><a href="#operating-system">Operating system</a></li>
      <li><a href="#required-programs">Required programs</a></li>
      <li><a href="#input-files">Input files</a></li>
      </ul>
</li>
      <li><a href="#installation">Installation</a></li>
      <li>
<a href="#preparing-for-badirate-run">Preparing for BadiRate run</a><ul class="nav nav-pills nav-stacked">
<li><a href="#splitting-family-size-file">Splitting family size file</a></li>
      <li><a href="#creating-job-scripts-and-setup-table">Creating job scripts and setup table</a></li>
      </ul>
</li>
      <li>
<a href="#running-badirate-jobs">Running BadiRate jobs</a><ul class="nav nav-pills nav-stacked">
<li><a href="#pbstorque">PBS/Torque</a></li>
      <li><a href="#slurm">SLURM</a></li>
      </ul>
</li>
      <li><a href="#collecting-outputs">Collecting outputs</a></li>
      <li>
<a href="#model-selection-and-gainloss-estimates">Model selection and gain/loss estimates</a><ul class="nav nav-pills nav-stacked">
<li><a href="#selecting-best-models-and-testing-its-significance">Selecting best models and testing its significance</a></li>
      <li><a href="#parsing-gainloss-events">Parsing gain/loss events</a></li>
      </ul>
</li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Gergo Palfalvi.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
